### PR TITLE
Fix bugs due to column to row changes for issue #290

### PR DESCRIPTION
**Pull request by junchaoxia**
Closing issue #290.

### CODE REVIEW

File: src/aspire/estimation/covar2d.py [hunk 1]
```diff
@@ -633,9 +633,9 @@ def get_cwf_coeffs(self, coeffs, ctf_fb, ctf_idx, mean_coeff, covar_coeff, noise
             mean_coeff_k = ctf_fb_k.apply(mean_coeff)
 
             coeff_est_k = coeff_k - mean_coeff_k
-            coeff_est_k = sig_noise_covar_coeff.solve(coeff_est_k)
+            coeff_est_k = sig_noise_covar_coeff.solve(coeff_est_k.T)
```
Comment (janden) [inline src/aspire/estimation/covar2d.py]: Add transpose here and remove it in line 638.

File: src/aspire/source/relion.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def load_single_mrcs(filepath, df):
 
             for future in futures.as_completed(to_do):
                 data_indices, data = future.result()
-                im[data_indices-start] = data[data_indices-start]
+                im[data_indices-start] = data
```
Comment (janden) [inline src/aspire/source/relion.py:167]: Why was there an index on `data` but no longer? This seems unrelated to row vs. column major.

**Review by janden — CHANGES_REQUESTED** [review]

File: src/aspire/source/relion.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/source/relion.py:167]: I think that was a legit bug that I introduced. I believe what Junchao has suggested for this line reflects how the original code was. I suspect the code path didn't exercise this difference. My bad (I expected at least a few bugs :) ).

Comment (janden) [reply src/aspire/source/relion.py:167]: Ok I see.

File: src/aspire/estimation/covar2d.py (hunk 1 above)
Comment (junchaoxia) [reply src/aspire/estimation/covar2d.py]: It looks that it need to transpose back again in line 637 for apply function and the transpose in 638 is still needed. This statements are similar to those in the base class.

Comment (janden) [reply src/aspire/estimation/covar2d.py]: This is because the `apply`/`solve` functions operate with a column-major convention. We may want to change this in the future, but the workaround right now should be to transpose the input and transpose the output for each `apply`/`solve` call to make that clear. As it is right now, sometimes `coeff_est_k` is row-major and sometimes it is column-major, which is confusing.

Comment (junchaoxia) [reply src/aspire/estimation/covar2d.py]: Added transpose in new commit 2cd4e2d.

Comment (janden) [reply src/aspire/estimation/covar2d.py]: I don't understand. Didn't you say that the input needed to be transposed again for `apply` on line 637?

Comment (junchaoxia) [reply src/aspire/estimation/covar2d.py]: I went through it again yesterday. I probably put additional T for input of solve or some where. The version in the commit should be OK.

Comment (janden) [reply src/aspire/estimation/covar2d.py]: I still don't understand. Why does the input of `solve` need to be transposed, but not the input of `apply`? Is this code being tested?

Comment (junchaoxia) [reply src/aspire/estimation/covar2d.py]: My original changes to these lines are using the same structures in lines 306 to 308 of the base class which have the unit test for cwf coefficients. I were able to run through the experiment example.  I will add the unit test in this PR to remove the confusion instead of in different PR.

Comment (junchaoxia) [reply src/aspire/estimation/covar2d.py]: Merged with the develop, changed to use row major for the cwf coefficients in all related statements, and added the unit test for issue #294 in the new commit df790ca.

Comment (janden) [reply src/aspire/estimation/covar2d.py]: Ok. I'm not sure how this ran before, but the code right now looks consistent.

**Review by janden — APPROVED** [review]
